wiki-webui/src/main/java/org/exoplatform/wiki/commons/Utils.java

```diff
@@ -546,4 +546,25 @@ public static String getDraftIdSessionKey() {
     return ConversationState.getCurrent().getIdentity().getUserId()
           + DRAFT_ID;
   }
+  
+  public static String getWikiTypeFromWikiId(String wikiId) {
+    String wikiType = "";
+    if (wikiId.startsWith("/spaces/")) {
+      wikiType = PortalConfig.GROUP_TYPE;
+    } else if (wikiId.startsWith("/user/")) {
+      wikiType = PortalConfig.USER_TYPE;
+    } else {
+      if (wikiId.startsWith("/")) {
+        wikiType = PortalConfig.PORTAL_TYPE;
+      }
+    }
+    return wikiType;
+  }
+
+  public static String getWikiOwnerFromWikiId(String wikiId) {
+    if(PortalConfig.USER_TYPE.equals(getWikiTypeFromWikiId(wikiId)) || PortalConfig.PORTAL_TYPE.equals(getWikiTypeFromWikiId(wikiId))) {

```

Comment:
If both conditions are checked, getWikiTypeFromWikiId is called twice. You should first assign getWikiTypeFromWikiId result to a variable, and then test :

String wikiType = getWikiTypeFromWikiId(wikiId);
if(PortalConfig.USER_TYPE.equals(wikiType) || PortalConfig.PORTAL_TYPE.equals(wikiType)) {
  ...